subsys/net/lib/azure_fota/azure_fota.c

```diff
@@ -176,20 +199,24 @@ static void fota_dl_handler(const struct fota_download_evt *fota_evt)
 }
 
 /* Checks the 'reported' object for current FOTA state
- * Returns true if the fota_state has been updated and should be reported to
+ * Returns true if the current_status has been updated and should be reported to
  * the device twin. If no action is needed, false is returned.
  */
 static bool parse_reported_status(const char *msg)
 {
 	struct cJSON *root_obj, *reported_obj, *fw_obj, *fw_status_obj,
-		*fw_current_obj, *fw_pending_obj;
+		*fw_current_obj, *fw_pending_obj, *job_id_obj;
 	bool report_needed = false;
 
 	/* The 'reported' object has the excpected structure if it exists:
```

Comment:
Add a comment here, that jobs are considered new if no jobId is present, and that this is so we can support older shadows without breaking.

Reply:
Claryfying (hopefully) comments added where the job ID object is used